### PR TITLE
[UNDERTOW-1675] Upgrade dependencies for 2.1.0.Final release

### PR DESCRIPTION
**Pull request by fl4via**
Jira: https://issues.redhat.com/browse/UNDERTOW-1675

### CODE REVIEW

**Comment by fl4via** [comment]
Retest this please

File: core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public void simpleRedirectTestCase() throws IOException, GeneralSecurityExceptio
             sendRequest(client, "/foo+bar", null);
             sendRequest(client, "/foo+bar;aa", null);
             sendRequest(client, "/foo+bar;aa", "x=y");
-            sendRequest(client, "/foo+bar%3Aaa", "x=%3Ablah");
+            sendRequest(client, "/foo+bar:aa", "x=%3Ablah");
```
Comment (jstourac) [inline core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java:88]: Why this change? Isn't this supposed to test https://issues.redhat.com/browse/UNDERTOW-1565?

Comment (fl4via) [reply core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java:88]: @jstourac  this is a result of the upgrade to Jakarta spec. I don't think it is a problem, because "/foo+bar:aa" is the output, and to be that's consistent to the remaining of the other expected results, but I might be mistaken.

Comment (fl4via) [reply core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java:88]: @jstourac  I figured what is causing this, I'll submit a fix pretty soon

Comment (fl4via) [reply core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java:88]: and, btw, you're right, thankfully the problem is in the test and not in Undertow

Comment (fl4via) [reply core/src/test/java/io/undertow/server/security/SimpleConfidentialRedirectTestCase.java]: @jstourac  done! The test now passes again

**Comment by fl4via** [comment]
Retest this please
